File: Project-Euler/Q3-Largest-Prime-Factor/Q3.java
/*
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600 851 475 143 ?
*/

import java.lang.*;

public class Q3 {
  public static void main(String[] args) {
    long lNum = 600851475143L;
    long lPrime = 0L;

    for (long i = 3L; i < (long)Math.sqrt(lNum); i += 2) {

      // If i is a factor, examine it
      if ((lNum % i) == 0) {

        // Check if i is prime
        for (long j = 2L; j <= (long)Math.sqrt(i); j++) {
          if ((i % j) == 0) { // If i is -not- prime
            break;
          } else if (((i % j) != 0) && ( j == (long)Math.sqrt(i))) { // If i -is- prime
            if (i > lPrime) {
              lPrime = i; // Change max prime if needed
              System.out.println("Largest Prime Factor So Far: " + lPrime);
            }
          }
        }
      }
    }
    System.out.println("Largest Prime Factor Found: " + lPrime);
  }
}

File: Project-Euler/Q3-Largest-Prime-Factor/Q3.py
# The prime factors of 13195 are 5, 7, 13 and 29.
#
# What is the largest prime factor of the number 600 851 475 143 ?

import time
import math

start_time = time.clock()

MAX_NUM = 600851475143
largest_prime, i = 2, 3

while i <= int(math.sqrt(MAX_NUM)):
    #print("i:", i)
    if MAX_NUM % i == 0:
        j = 2
        while j <= int(math.sqrt(i)):
            if i % j == 0:
                break
            elif i % j != 0 and j == int(math.sqrt(i)):
                if i > largest_prime:
                    largest_prime = i
            j += 1
    i += 2

print("Largest Prime:", largest_prime)
print("\nRuntime:", (time.clock() - start_time) * 1000, "ms")
